sympy/core/numbers.py

```diff
@@ -155,26 +155,32 @@ def igcd(*args):
     5
 
     """
-    a = args[0]
-    for b in args[1:]:
+    a = 0
+    args = list(args)
+    while args:
+        a = args.pop()
+        if a == 0:
+            continue
+        break
+    a = abs(as_int(a))
+    while args:
+        b = args.pop()
+        if b == 0:
+            continue
+        b = as_int(b)
+        t = a, b
         try:
-            a = _gcdcache[(a, b)]
+            a = _gcdcache[t]
         except KeyError:
-            a, b = as_int(a), as_int(b)
-
-            if a and b:
-                if b < 0:
-                    b = -b
-
-                while b:
-                    a, b = b, a % b
-            else:
-                a = abs(a or b)
-
-            _gcdcache[(a, b)] = a
-        if a == 1 or b == 1:
-            return 1
-    return a
+            while b:
+                a, b = b, a % b
+            if a < 0:
+                a = -a
+            if a == 1:
+                test_others = [as_int(a) for a in args]
+                return 1
+            _gcdcache[t] = a
```

Comment:
I was thinking not to cache the trivial arg but `a` is now the result. You're right. I'll move it.